qCC/ccCommandLineCommands.cpp

```diff
@@ -1348,19 +1349,114 @@ bool CommandSubsample::process(ccCommandLineInterface& cmd)
 		{
 			return cmd.error(QObject::tr("Missing parameter: spatial step after \"-%1 SPATIAL\"").arg(COMMAND_SUBSAMPLE));
 		}
+
 		bool ok;
 		double step = cmd.arguments().takeFirst().toDouble(&ok);
 		if (!ok || step <= 0)
 		{
-			return cmd.error(QObject::tr("Invalid step value for spatial resampling!"));
+			return cmd.error(QObject::tr("Invalid step value for spatial subsampling!"));
 		}
 		cmd.print(QObject::tr("\tSpatial step: %1").arg(step));
-		
+
+		double sfMinSpacing = 0;
+		double sfMaxSpacing = 0;
+		bool useActiveSF = false;
+		if (!cmd.arguments().empty())
+		{
+			if (cmd.arguments().front().toUpper() == OPTION_USE_ACTIVE_SF)
+			{
+				//enable USE_ACTIVE_SF
+				useActiveSF = true;
+				cmd.arguments().pop_front();
+				if (cmd.arguments().size() >= 2)
+				{
+					bool validMin;
+					sfMinSpacing = cmd.arguments().takeFirst().toDouble(&validMin);
+					bool validMax;
+					sfMaxSpacing = cmd.arguments().takeFirst().toDouble(&validMax);
+					if (!(validMin&&validMax) || sfMinSpacing < 0 || sfMaxSpacing < 0)
+					{
+						return cmd.error(QObject::tr("Invalid parameters: Two positive decimal number required after '%1'").arg(OPTION_USE_ACTIVE_SF));
+					}
+				}
+				else
+				{
+					return cmd.error(QObject::tr("Missing parameters: Two positive decimal number required after '%1'").arg(OPTION_USE_ACTIVE_SF));
+				}
+			}
+		}
+
 		for (CLCloudDesc& desc : cmd.clouds())
 		{
 			cmd.print(QObject::tr("\tProcessing cloud %1").arg(!desc.pc->getName().isEmpty() ? desc.pc->getName() : "no name"));
-			
+
 			CCCoreLib::CloudSamplingTools::SFModulationParams modParams(false);
+
+			//handle Use Active SF on each cloud
+			if (useActiveSF)
+			{
+				//look for the min and max sf values
+				ccScalarField* sf = desc.pc->getCurrentDisplayedScalarField();
```

Comment:
So I had some doubts about this when looking at the code yesterday: since we are in the command line, I can't guarantee that there's always a 'displayed' scalar field active. It would be safer to use 'getCurrentInScalarFieldIndex' instead.

I see that there may be similar issues with other tools, I'll change that later.

Reply:
but that is gonna be an issue inside the called subsampling tools as well, because it calls the same displayedscalarfield.

So better would be to just make sure there is a displayed scalar field?
In this case
desc.pc->setDisplayedScalarField(index)

Reply:
Oh, I see it's a big mess, as normally the displayed scalar field should be the 'out' scalar field in fact... (not the 'in'). But that's not consistent with how tools should work. Typically the subsampling tool should look at the 'in' scalar field 😅 (which should be properly set before calling the tool).

I guess you can forget about this for now then. I'll have to make a big cleanup pass later.

Reply:
I'll keep that in my mind for later :)